util/buildinfo/buildinfo.go

```diff
@@ -1,3 +1,5 @@
+// Deprecated: Build information format has been deprecated and will be removed
+// in a future release. Use provenance attestations instead.
```

Comment:
For deprecation in _code_, I try to have a (docs) link to the replacement package. I guess in this case that may be hard if the replacement is not a "1:1" replacement.


If the package has a `doc.go`, it would work to have that `doc.go` both a `Package foo does bar` description (package doc in the right format) _and_ the deprecated comment;

```go
// Package buildinfo provides ....
//
// Deprecated: the buildinfo package is deprecated in favour of bilabial. Use [github.com/foo/bar] instead.
```

If it's not a drop-in replacement, then perhaps we need something documented somewhere how to migrate (although it depends on how many use the old feature, not sure how _much_ time to spend on it).

Reply:
It was mostly an internal feature but has been replaced in favor of attestation provenance. Don't think there is a migration path atm. Also the `buildinfo` package is an utility one and not really tight to the client. Our docs with the disclaimer on top should be enough I think  https://github.com/crazy-max/buildkit/blob/deprecate-buildinfo/docs/buildinfo.md?

Reply:
Created a dedicated page like the one on docker/cli. Let me know if that SGTY.